core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1539,6 +1539,13 @@ abstract class RDD[T: ClassTag](
     // NOTE: we use a global lock here due to complexities downstream with ensuring
     // children RDD partitions point to the correct parent partitions. In the future
     // we should revisit this consideration.
+    if (doCheckpointCalled) {
+      throw new SparkException("Because job has been executed on this RDD, checkpoint won't work")
+    }
```

Comment:
All right, how about change to following code:
```
if (doCheckpointCalled) {
      logWarning(s"Because job has been executed on RDD ${id}, checkpoint won't work")
}
```